src/core/net/ip6.cpp

```diff
@@ -504,6 +548,8 @@ ThreadError Ip6::HandleExtensionHeaders(Message &message, Header &header, uint8_
             break;
 
         case kProtoIp6:
+            ExitNow();
```

Comment:
It might be architecturally cleaner to actually strip the outer IPv6 header and run the packet through `HandleDatagram()` again as if it was a newly received datagram.